build_script.py

```diff
@@ -36,146 +37,218 @@ def _mkdirp(path):
-            core_foundation_build_dir=core_foundation_build_dir,
-            lit_path=lit_path,
-            lit_flags=lit_flags,
-            tests_path=tests_path))
-
-
-def _install(args):
-    """
-    Install the XCTest.so, XCTest.swiftmodule, and XCTest.swiftdoc build
-    products into the given module and library paths.
-    """
-    build_dir = os.path.abspath(args.build_dir)
-    module_install_path = os.path.abspath(args.module_install_path)
-    library_install_path = os.path.abspath(args.library_install_path)
-
-    _mkdirp(module_install_path)
-    _mkdirp(library_install_path)
-
-    xctest_so = "libXCTest.so"
-    run("cp {} {}".format(
-        os.path.join(build_dir, xctest_so),
-        os.path.join(library_install_path, xctest_so)))
-
-    xctest_swiftmodule = "XCTest.swiftmodule"
-    run("cp {} {}".format(
-        os.path.join(build_dir, xctest_swiftmodule),
-        os.path.join(module_install_path, xctest_swiftmodule)))
-
-    xctest_swiftdoc = "XCTest.swiftdoc"
-    run("cp {} {}".format(
-        os.path.join(build_dir, xctest_swiftdoc),
-        os.path.join(module_install_path, xctest_swiftdoc)))
+class DarwinStrategy:
+    @staticmethod
+    def requires_foundation_build_dir():
+        """
+        The Foundation build directory is not required on Darwin because the
+        Xcode workspace implicitly builds Foundation when building the XCTest
+        schemes.
+        """
+        return False
+
+    @staticmethod
+    def build(args):
+        """
+        Build XCTest and place the built products in the given 'build_dir'.
+        If 'test' is specified, also executes the 'test' subcommand.
+        """
+        swiftc = os.path.abspath(args.swiftc)
+        build_dir = os.path.abspath(args.build_dir)
+
+        run("xcodebuild -workspace {source_dir}/XCTest.xcworkspace "
+            "-scheme SwiftXCTest "
+            "SWIFT_EXEC=\"{swiftc}\" "
+            "SWIFT_LINK_OBJC_RUNTIME=YES "
+            "SYMROOT=\"{build_dir}\" OBJROOT=\"{build_dir}\"".format(
+                swiftc=swiftc,
+                build_dir=build_dir,
+                source_dir=SOURCE_DIR))
+
+        if args.test:
+            # Execute main() using the arguments necessary to run the tests.
+            main(args=["test",
+                       "--swiftc", swiftc,
+                       build_dir])
+
+    @staticmethod
+    def test(args):
+        """
+        Test SwiftXCTest.framework, using the given 'swiftc' compiler, looking
+        for it in the given 'build_dir'.
+        """
+        swiftc = os.path.abspath(args.swiftc)
+        build_dir = os.path.abspath(args.build_dir)
+
+        run("xcodebuild -workspace {source_dir}/XCTest.xcworkspace "
+            "-scheme SwiftXCTestFunctionalTests "
+            "SWIFT_EXEC=\"{swiftc}\" "
+            "SWIFT_LINK_OBJC_RUNTIME=YES "
+            "SYMROOT=\"{build_dir}\" OBJROOT=\"{build_dir}\" "
+            "| grep -v \"    export\"".format(
```

Comment:
Oh, good eye @modocache! I meant to call that out in the PR description for discussion, but forgot.

The purpose of that grep was to filter out several dozen lines of output that Xcode emits, showing the complete environment that is being exported when it invokes lit as an External Build Tool. I was finding that this output was more distracting than useful when I was using the script to repeatedly run tests while working on XCTest. They said, I'm not particularly attached to this behavior and would have no issue with leaving it out. What do you think?